# Retail Intelligence Business Analytics

## Notebook 1: Data Understanding

### Objective

The objective of this notebook is to explore the Global Superstore dataset, understand its structure, examine key attributes, and perform an initial assessment before performing data quality assessment, cleaning, feature engineering, and business analysis.

In [1]:
#Importing Libraries:

import pandas as pd
import numpy as np

# Display all columns
pd.set_option("display.max_columns", None)

In [2]:
df = pd.read_excel(
    "../data/raw/Global Superstore Data.xlsx",
    sheet_name="Orders"
)

## Step 1: Preview the Dataset

Display the first few records of the dataset to verify that it has been loaded correctly and to gain an initial understanding of its structure and contents.

In [3]:
df.head()

,Row ID,Order ID,Order Date,Ship Date,Ship Mode,Customer ID,Customer Name,Segment,Postal Code,City,State,Country,Region,Market,Product ID,Product Name,Sub-Category,Category,Sales,Quantity,Discount,Profit,Shipping Cost,Order Priority
0,24599,IN-2017-CA120551-42816,2017-03-22,2017-03-29,Standard Class,CA-120551,Cathy Armstrong,Home Office,NaN,Herat,Hirat,Afghanistan,Southern Asia,Asia Pacific,FUR-BO-4861,"Ikea Library with Doors, Mobile",Bookcases,Furniture,731.82,2,0.0,102.42,39.66,Medium
1,29465,ID-2015-BD116051-42248,2015-09-01,2015-09-04,Second Class,BD-116051,Brian Dahlen,Consumer,NaN,Herat,Hirat,Afghanistan,Southern Asia,Asia Pacific,OFF-SU-2988,"Acme Scissors, Easy Grip",Supplies,Office Supplies,243.54,9,0.0,104.49,18.72,Medium
2,24598,IN-2017-CA120551-42816,2017-03-22,2017-03-29,Standard Class,CA-120551,Cathy Armstrong,Home Office,NaN,Herat,Hirat,Afghanistan,Southern Asia,Asia Pacific,TEC-MA-4211,"Epson Receipt Printer, White",Machines,Technology,346.32,3,0.0,13.77,14.10,Medium
3,24597,IN-2017-CA120551-42816,2017-03-22,2017-03-29,Standard Class,CA-120551,Cathy Armstrong,Home Office,NaN,Herat,Hirat,Afghanistan,Southern Asia,Asia Pacific,FUR-FU-5726,"Rubbermaid Door Stop, Erganomic",Furnishings,Furniture,169.68,4,0.0,79.68,11.01,Medium
4,29464,ID-2015-BD116051-42248,2015-09-01,2015-09-04,Second Class,BD-116051,Brian Dahlen,Consumer,NaN,Herat,Hirat,Afghanistan,Southern Asia,Asia Pacific,OFF-EN-3664,"Cameo Interoffice Envelope, with clear poly wi...",Envelopes,Office Supplies,203.88,4,0.0,24.36,5.72,Medium


In [4]:
df.shape

(51290, 24)

In [5]:
df.columns.tolist()

['Row ID',
 'Order ID',
 'Order Date',
 'Ship Date',
 'Ship Mode',
 'Customer ID',
 'Customer Name',
 'Segment',
 'Postal Code',
 'City',
 'State',
 'Country',
 'Region',
 'Market',
 'Product ID',
 'Product Name',
 'Sub-Category',
 'Category',
 'Sales',
 'Quantity',
 'Discount',
 'Profit',
 'Shipping Cost',
 'Order Priority']

## Step 2: Inspect Dataset Structure

The `info()` method provides a summary of the dataset, including:

- Number of rows and columns
- Data types
- Non-null values
- Memory usage

This helps identify missing values and ensures that each column has the expected data type.

In [6]:
df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 51290 entries, 0 to 51289
Data columns (total 24 columns):
 #   Column          Non-Null Count  Dtype         
---  ------          --------------  -----         
 0   Row ID          51290 non-null  int64         
 1   Order ID        51290 non-null  object        
 2   Order Date      51290 non-null  datetime64[ns]
 3   Ship Date       51290 non-null  datetime64[ns]
 4   Ship Mode       51290 non-null  object        
 5   Customer ID     51290 non-null  object        
 6   Customer Name   51290 non-null  object        
 7   Segment         51290 non-null  object        
 8   Postal Code     9994 non-null   float64       
 9   City            51290 non-null  object        
 10  State           51290 non-null  object        
 11  Country         51290 non-null  object        
 12  Region          51290 non-null  object        
 13  Market          51290 non-null  object        
 14  Product ID      51290 non-null  object        
 15  Pr

In [7]:
df.sample(5, random_state=42)

,Row ID,Order ID,Order Date,Ship Date,Ship Mode,Customer ID,Customer Name,Segment,Postal Code,City,State,Country,Region,Market,Product ID,Product Name,Sub-Category,Category,Sales,Quantity,Discount,Profit,Shipping Cost,Order Priority
49728,39143,CA-2015-JK15205140-42274,2015-09-27,2015-10-02,Standard Class,JK-152051408,Jamie Kunitz,Consumer,22204.0,Arlington,Virginia,United States,Southern US,USCA,TEC-PH-3811,Cisco SPA301,Phones,Technology,1871.880,12,0.0,561.5640,117.95,Medium
45547,36842,CA-2017-MG17890140-43001,2017-09-23,2017-09-27,Standard Class,MG-178901406,Michael Granlund,Home Office,3301.0,Concord,New Hampshire,United States,Eastern US,USCA,OFF-ST-6266,Tenex Personal Project File with Scoop Front D...,Storage,Office Supplies,67.400,5,0.0,17.5240,6.11,Medium
15664,11413,ES-2017-PT1909048-42954,2017-08-07,2017-08-10,First Class,PT-1909048,Pete Takahito,Consumer,NaN,Aschaffenburg,Bavaria,Germany,Western Europe,Europe,TEC-PH-5271,"Motorola Speaker Phone, Full Size",Phones,Technology,655.350,5,0.0,72.0000,108.11,Medium
40561,35001,CA-2015-DH13075140-42168,2015-06-13,2015-06-19,Standard Class,DH-130751408,Dave Hallsten,Corporate,35601.0,Decatur,Alabama,United States,Southern US,USCA,TEC-CO-3590,Brother DCP1000 Digital 3 in 1 Multifunction M...,Copiers,Technology,899.970,3,0.0,314.9895,73.62,Medium
49426,39371,CA-2017-JM15250140-42737,2017-01-02,2017-01-06,Standard Class,JM-152501402,Janet Martin,Consumer,77340.0,Huntsville,Texas,United States,Central US,USCA,OFF-BI-5640,"Pressboard Covers with Storage Hooks, 9 1/2"" x...",Binders,Office Supplies,13.748,14,0.8,-22.6842,1.90,Medium


## Step 3: Summary Statistics

Generate descriptive statistics for numerical columns to understand the distribution of the data and identify any unusual values.

In [8]:
df.describe()

,Row ID,Order Date,Ship Date,Postal Code,Sales,Quantity,Discount,Profit,Shipping Cost
count,51290.00000,51290,51290,9994.000000,51290.000000,51290.000000,51290.000000,51290.000000,51290.000000
mean,25645.50000,2016-05-11 11:52:18.436342272,2016-05-15 11:09:41.306297344,55190.379428,246.490581,3.476545,0.142908,28.610982,26.478567
min,1.00000,2014-01-01 00:00:00,2014-01-03 00:00:00,1040.000000,0.444000,1.000000,0.000000,-6599.978000,1.002000
25%,12823.25000,2015-06-19 00:00:00,2015-06-23 00:00:00,23223.000000,30.758625,2.000000,0.000000,0.000000,2.610000
50%,25645.50000,2016-07-08 00:00:00,2016-07-12 00:00:00,56430.500000,85.053000,3.000000,0.000000,9.240000,7.790000
75%,38467.75000,2017-05-22 00:00:00,2017-05-26 00:00:00,90008.000000,251.053200,5.000000,0.200000,36.810000,24.450000
max,51290.00000,2017-12-31 00:00:00,2018-01-07 00:00:00,99301.000000,22638.480000,14.000000,0.850000,8399.976000,933.570000
std,14806.29199,NaN,NaN,32063.693350,487.565361,2.278766,0.212280,174.340972,57.251373


## Step 4: Explore Categorical Columns

Display summary statistics for categorical variables to understand the variety of values present in the dataset.

In [9]:
df.describe(include="object")

,Order ID,Ship Mode,Customer ID,Customer Name,Segment,City,State,Country,Region,Market,Product ID,Product Name,Sub-Category,Category,Order Priority
count,51290,51290,51290,51290,51290,51290,51290,51290,51290,51290,51290,51290,51290,51290,51290
unique,25728,4,17415,796,3,3650,1106,165,23,5,3788,3788,17,3,4
top,CA-2017-SV20365140-42999,Standard Class,SV-203651406,Muhammed Yedwab,Consumer,New York City,California,United States,Western Europe,Asia Pacific,OFF-FA-6129,Staples,Binders,Office Supplies,Medium
freq,14,30775,26,108,26518,915,2001,9994,5883,14302,227,227,6146,31289,29433


## Conclusion

The initial exploration confirms that the Global Superstore dataset has been loaded successfully and contains sufficient information for comprehensive business analysis.

Key observations:

- The dataset contains 51,290 records and 24 original features.
- Both numerical and categorical variables are present.
- Order and shipping dates have been imported correctly.
- Sales, Profit, Quantity, and Discount are available for performance analysis.
- Customer, Product, and Geographic information will support segmentation and regional analysis.

The next notebook focuses on assessing data quality by identifying missing values, duplicates, inconsistent records, and potential data issues before feature engineering.